uk_tax_23-24.py

```diff
@@ -80,16 +83,27 @@ def national_insurance(incomes):
 
     return ni
 
-def student_loan_repayment_plan_2(incomes):
-    repayments = np.where(
-        incomes > student_loan_plan_2_threshold,
-        (incomes - student_loan_plan_2_threshold) * student_loan_plan_2_rate,
-        0,
-    )
-
+# Can expand for further student loan repayment plans (3/4/5) in future
+def student_loan_repayment_plan(incomes, student_loan_plan_number=2):
+    if student_loan_plan_number == 1:
```

Comment:
With the dictionary, we can avoid the nested ifs/elseifs:
```
def student_loan_repayment_plan(incomes, student_loan_plan="Plan 2"):

  student_loan_threshold = student_loan_thresholds[student_loan_plan]

  if student_loan_plan == "Postgraduate":
    student_loan_rate = 6
  else:
    student_loan_rate = 9

  repayments = np.where(
    incomes > student_loan_threshold,
      (incomes - student_loan_threshold) * (student_loan_rate / 100),
      0,
  )

  return repayments
```

I would also like to avoid printing anything in the console in this function, the top level function does that already. This function will be used to calculate the repayments for all salaries in the range and we don't want that printing each time.